Assets/Scripts/Entity/Player/Player.cs

```diff
@@ -71,6 +71,29 @@ private void OnCollisionEnter(Collision other)
         // Don't bother doing anything if we can't take damage anyway
         if (invincible) return;
 
+        // Detecting projectiles
+        if (other.gameObject.CompareTag("EnemyProjectile"))
+        {
+            Projectile projectile = other.gameObject.GetComponent<Projectile>();
+            ChangeHealthBy(projectile.damage);
+            Destroy(other.gameObject);
+        }
+        else if (other.gameObject.CompareTag("Heart"))
```

Comment:
Like the projectile damage, this should go on the heart. Otherwise this player would be checking for every possible object that has a tag and is intractable. As opposed to the items themselves (projectile, meleeweapon, heart) simply checking if they hit the player or enemy and apply damage accordingly.